sys-block/gpart/patches/gpart-0.3.patchset.2

```diff
@@ -0,0 +1,139 @@
+From 147f08ed764eceb8cba4fe03e595b3a8dc7a09dd Mon Sep 17 00:00:00 2001
+From: My Name <mymail@mydomain.org>
+Date: Mon, 1 Jan 2018 03:35:25 +0000
+Subject: applying patch gpart-0.3.patchset.2
+
+
+diff --git a/src/disku.c b/src/disku.c
+index d50cebf..1761210 100644
+--- a/src/disku.c
++++ b/src/disku.c
+@@ -33,6 +33,11 @@
+ #include <sys/disk.h>
+ #endif
+ 
++#if defined(__HAIKU__)
++#include <drivers/Drivers.h>
++#include <fcntl.h>
++#endif
++
+ #include <unistd.h>
+ 
+ static void geometry_from_num_sectors(struct disk_geom *g, uint64_t nsects)
+@@ -95,8 +100,29 @@ static void os_disk_geometry(disk_desc *d, struct disk_geom *g)
+ 	g.d_nsecs = o / u;
+ 	g.d_c = g.d_nsecs / g.d_h / g.d_s;
+ }
++#elif defined(__HAIKU__)
++static void os_disk_geometry(disk_desc *d, struct disk_geom *g)
++{
++	device_geometry geom;
++	int fd;
++
++	if ((fd = open(d->d_dev, O_RDONLY)) < 0)
```

Comment:
Looks like an `fd` is already available at `d->d_fd`, so you should avoid having more side effects